src/store/data/operations/dataOperation.js

```diff
@@ -0,0 +1,28 @@
+import { createLogic } from 'redux-logic'
+import { REQUEST_DATA } from '../types'
+import { requestDataSuccess, requestDataStart } from '../actions'
+import { addMovies } from '../../movies/actions'
+
+export const dataOperation = createLogic({
+  type: REQUEST_DATA,
+  latest: true,
+  async process({ action, axios }, dispatch, done) {
+    dispatch(requestDataStart(true))
+    const { endpoint, url, schema } = action
+    try {
+      const dataRequest = await axios.get(url)
+      const { data } = dataRequest
+      const { result, entities } = schema(data.results)
+      const movies = {
+        items: result,
+        totalPages: data.total_pages,
+        currentPage: data.page
+      }
+      dispatch(addMovies(movies, endpoint))
+      dispatch(requestDataSuccess(entities.movies))
+    } catch (err) {
+      console.log('error')
```

Comment:
such things can be present only in 'work in progress' code. please remove debug code before review